### PR TITLE
don't represent site plugins with 'null' anymore

### PR DESCRIPTION
**Pull request by rmuir**
Lots of plugin logic manipulates Tuple<PluginInfo,Plugin> and historically site-only plugins used `null` for the Plugin part. This only makes for bugs: we can just have a simple Plugin implementation for these site plugins.

The bigger problem is we have no real site plugins in our build. This adds `example-site` plugin with a simple integration test.

### CODE REVIEW

File: plugins/example-site/src/site/_site/index.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<html>
+  <head>
+    <title>Page title</title>
+  </head>
+  <body>Page body</body>
```
Comment (jaymode) [inline plugins/example-site/src/site/_site/index.html]: can we add a simple integration test that just checks the content returned matches `Page body`?

File: distribution/zip/pom.xml [hunk 2]
```diff
@@ -13,7 +13,6 @@
     <name>Elasticsearch ZIP Distribution</name>
 
     <properties>
```
Comment (jaymode) [inline distribution/zip/pom.xml]: minor, but should we just remove the properties tags as well?

**Comment by jaymode** [comment]
left two small comments. I don't think adding the other integration test should block the PR. This is a big improvement. LGTM

File: plugins/example-site/src/site/_site/index.html (hunk 1 above)
Comment (rmuir) [inline plugins/example-site/src/site/_site/index.html]: I dont know how to do it. I'm a dummy. @clintongormley made these simple rest tests for the cloud plugins that just checked they were installed, and i used that logic here so we have something.

File: distribution/zip/pom.xml (hunk 2 above)
Comment (rmuir) [inline distribution/zip/pom.xml]: I was keeping those for all the distributions, we can remove those properties sections if we want. ill nuke em.

**Comment by rmuir** [comment]
I'm going to open with a followup to improve the rest test here to also try to check for the contents: its a really good idea and might be an easy task for someone more familiar with rest tests.

**Pull request merged**